go/src/koding/klient/machine/index/node.go

```diff
@@ -240,24 +254,31 @@ func (nd *Node) DiskSize(maxsize int64) (size int64) {
 //
 // It ignored nodes marked as deleted.
 func (nd *Node) ForEach(fn func(string, *Entry)) {
+	nd.forEach(fn, false)
+}
+
+// ForEachAll traverses the tree and calls fn on every node's entry.
+func (nd *Node) ForEachAll(fn func(string, *Entry)) {
```

Comment:
yes, sometimes I need to include deleted/virtual nodes and sometimes not. Since there is only one index we need to know which files are available locally(have no promises or promise add).

Also `index.Merge` needs to traverse all nodes even these marked as deleted.